Description: Upstream changes introduced in version 1.6.0-1ppa1~lucid1
 This patch has been created by dpkg-source during the package build.
 Here's the last changelog entry, hopefully it gives details on why
 those changes were made:
 .
 openfoam-1.6-ext (1.6.0-1ppa1~lucid1) lucid; urgency=low
 .
   * Initial build for lucid.
 .
 The person named in the Author field signed this changelog entry.
Author: Oliver Borm <oli.borm@web.de>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- openfoam-1.6-ext-1.6.0.orig/Allwmake
+++ openfoam-1.6-ext-1.6.0/Allwmake
@@ -14,7 +14,7 @@ fi
 
 # build ThirdParty sources
 #( cd $WM_THIRD_PARTY_DIR && ./Allwmake )
-( cd $WM_THIRD_PARTY_DIR && ./AllMake )
+# ( cd $WM_THIRD_PARTY_DIR && ./AllMake )
 
 # build OpenFOAM libraries and applications
 src/Allwmake
--- openfoam-1.6-ext-1.6.0.orig/bin/paraFoam
+++ openfoam-1.6-ext-1.6.0/bin/paraFoam
@@ -1,133 +1,148 @@
 #!/bin/sh
-#------------------------------------------------------------------------------
-# =========                 |
-# \\      /  F ield         | OpenFOAM: The Open Source CFD Toolbox
-#  \\    /   O peration     |
-#   \\  /    A nd           | Copyright held by original author
-#    \\/     M anipulation  |
-#-------------------------------------------------------------------------------
-# License
-#     This file is part of OpenFOAM.
-#
-#     OpenFOAM is free software; you can redistribute it and/or modify it
-#     under the terms of the GNU General Public License as published by the
-#     Free Software Foundation; either version 2 of the License, or (at your
-#     option) any later version.
-#
-#     OpenFOAM is distributed in the hope that it will be useful, but WITHOUT
-#     ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-#     FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
-#     for more details.
-#
-#     You should have received a copy of the GNU General Public License
-#     along with OpenFOAM; if not, write to the Free Software Foundation,
-#     Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+#
+# Copyright (c) 2008-2009 Takuya OSHIMA <oshima@eng.niigata-u.ac.jp>.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met:
+#
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the
+#    distribution.
+#
+#  * Neither the name of the author nor the names of any contributors
+#    may be used to endorse or promote products derived from this
+#    software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #
 # Script
-#     paraFoam
+#     pvFoam
 #
 # Description
-#     start paraview with the OpenFOAM libraries
-#
-#------------------------------------------------------------------------------
-usage() {
-    while [ "$#" -ge 1 ]; do echo "$1"; shift; done
-    cat<<USAGE
-
-usage: ${0##*/} [OPTION]
-options:
-  -case   dir    specify alternative case directory
-  -region name   specify mesh region name
-  -touch         only create the .OpenFOAM file
-
-* start paraview $ParaView_VERSION with the OpenFOAM libraries
+#     ParaView interface script for use with vtkPOpenFOAMReader.
 
-USAGE
+# Adjust as necessary.
+# You might also want to set LD_LIBRARY_PATH or DYLD_LIBRARY_PATH.
+paraviewPath="`which paraview`"
+# or in full path like paraviewPath=/opt/ParaView3/bin/paraview
+
+if [ ! -x "${paraviewPath}" -o -d "${paraviewPath}" ]; then
+    # not using LINENO which is not supported by dash (yet)
+    echo "ParaView not found; set paraviewPath at line" `awk '/paraviewPath/{print FNR;exit 0}' $0` "of $0."
     exit 1
-}
+fi
 
-unset regionName touchOnly
+ext=foam
+casePath=.
+script=${0##*/}
+background=yes
+trapCommand=trap
 
-# parse options
-while [ "$#" -gt 0 ]
+while [ $# -gt 0 ]
 do
     case "$1" in
-    -h | -help)
-        usage
-        ;;
-    -case)
-        [ "$#" -ge 2 ] || usage "'$1' option requires an argument"
-        cd "$2" 2>/dev/null || usage "directory does not exist:  '$2'"
-        shift 2
-        ;;
-    -region)
-        [ "$#" -ge 2 ] || usage "'$1' option requires an argument"
-        regionName=$2
-        shift 2
-        ;;
-    -touch)
-        touchOnly=true
-        shift
-        ;;
-    *)
-        usage "unknown option/argument: '$*'"
-        ;;
+	-case)
+	    if [ $# -lt 2 ]; then
+		echo "-case requires path specification"
+		exit 1
+	    fi
+	    shift
+	    casePath="$1"
+	    ;;
+	-clientServer|-cs)
+	    if [ $# -lt 2 ]; then
+		echo "-clientServer or -cs requires specification of the server resource name"
+		exit 1
+	    fi
+	    shift
+	    csResourceName="$1"
+	    ;;
+	-foreground|-fg)
+	    background=no
+	    ;;
+	-keepStub|-ks)
+	    trapCommand=:
+	    ;;
+	-help)
+	    echo "Usage: ${script} [-case <dir>] [-clientServer|-cs <server resource name>] [-foreground|-fg] [-keepStub|-ks] [-help]"
+	    exit 0
+	    ;;
     esac
+    shift
 done
 
-# get a sensible caseName
-caseName=${PWD##*/}
-caseFile="$caseName.OpenFOAM"
-fvControls="system"
-
-if [ -n "$regionName" ]
-then
-    caseFile="$caseName{$regionName}.OpenFOAM"
-    fvControls="$fvControls/$regionName"
-fi
-
-if [ -n "$touchOnly" ]
-then
-    touch "$caseFile"
-    echo "created '$caseFile'"
-    exit 0
+# cd to the case directory
+if ! cd "${casePath}" > /dev/null 2>&1; then
+    echo "Cannot change directory to \"${casePath}\"; exiting"
+    exit 1
 fi
 
-# parent directory for normal or parallel results
-case "$caseName" in
-    processor*) parentDir=".." ;;
-    *)          parentDir="."  ;;
-esac
-
-# check existence of essential files
-for check in system/controlDict $fvControls/fvSchemes $fvControls/fvSolution
-do
-    [ -s "$parentDir/$check" ] || usage "file does not exist:  '$parentDir/$check'"
-done
+caseName="${PWD##*/}"
 
+# see if caseName starts with controlDict or the case is in the root directory.
+if [ "${caseName#controlDict}" != "${caseName}" -o ! "${caseName}" ]; then
+    # Use case.foam as file name given to ParaView
+    caseName="case"
+fi
 
-case "$ParaView_VERSION" in
-2*)
-    trap "rm -f paraFoam.pvs $caseFile 2>/dev/null; exit 0" EXIT TERM INT
-    touch "$caseFile"
-
-    # since we are now in the cwd, %CASE% is '$PWD/$caseFile'
-    sed -e s@%CASE%@$PWD/$caseFile@g \
-        $WM_PROJECT_DIR/bin/tools/paraFoam.pvs > paraFoam.pvs
-
-    paraview paraFoam.pvs
-    ;;
-
-*)
-    # only create/remove caseFile if it didn't already exist
-    [ -e $caseFile ] || {
-        trap "rm -f $caseFile 2>/dev/null; exit 0" EXIT TERM INT
-        touch "$caseFile"
-        echo "created temporary '$caseFile'"
-    }
+fileName="${caseName}.${ext}"
+if [ -e "${fileName}" ]; then
+    if [ ! -r "${fileName}" ]; then
+	echo "Stub ${fileName} exists but not readable; exiting"
+	exit 1
+    fi
+    trapCommand=:
+else
+    if ! touch "${fileName}" > /dev/null 2>&1; then
+	echo "Cannot create stub ${fileName}; exiting"
+	exit 1
+    fi
+fi
 
-    paraview --data="$caseFile"
-    ;;
+# set the locale where ParaView is run under to "C" so that the
+# interpretation of decimal points by the system routines are not
+# affected
+export LANG=C
+export LC_ALL=C
+
+# specify fileName with full path so that it can easily be accessed
+# later with File->Recent Files
+if [ "${background}" = "yes" ]; then
+    (
+	${trapCommand} "rm -f \"${fileName}\" ; exit 0" \
+	    EXIT HUP INT PIPE ALRM TERM USR1 USR2 ;
+	if [ "${csResourceName}" ]; then
+	    "${paraviewPath}" --data="${PWD}/${fileName}" \
+		--server="${csResourceName}"
+	else
+	    "${paraviewPath}" --data="${PWD}/${fileName}"
+	fi
+    ) &
+else
+    ${trapCommand} "rm -f \"${fileName}\" ; exit 0" \
+	EXIT HUP INT PIPE ALRM TERM USR1 USR2
+    if [ "${csResourceName}" ]; then
+	"${paraviewPath}" --data="${PWD}/${fileName}" \
+	    --server="${csResourceName}"
+    else
+	"${paraviewPath}" --data="${PWD}/${fileName}"
+    fi
+fi
 
-esac
 #------------------------------------------------------------------------------
--- /dev/null
+++ openfoam-1.6-ext-1.6.0/bin/createDebianSourcePackage.sh
@@ -0,0 +1,24 @@
+#!/bin/sh
+
+# TODO: checkout correct tag
+GPGKey=${1}
+ubuntuVersion="10.04"
+myTag="1.6-extRelease"
+pushOrigSources="-sa" ## yes
+# pushOrigSources="-sd" ## no
+
+## create debian changelog for a release
+# git-dch --release
+## or for a snapshot
+# git-dch --snapshot
+
+
+## create debian source package with *.orig.tar.gz and patches
+git-buildpackage \
+	--git-builder="debuild -i\.git -I.git -S ${pushOrigSources} -k${GPGKey}" \
+	--git-upstream-branch=master \
+	--git-debian-branch=packaging/ubuntu/${ubuntuVersion} \
+	--git-ignore-new
+
+## build debian source package in pbuilder
+# sudo pbuilder build --pkgname-logfile ../openfoam-1.6-ext_1.6.0-1ppa1~lucid1.dsc
--- openfoam-1.6-ext-1.6.0.orig/wmake/Makefile
+++ openfoam-1.6-ext-1.6.0/wmake/Makefile
@@ -58,7 +58,7 @@ WM_SCRIPTS   = $(WM_DIR)/scripts
 
 LIB_SRC            = $(WM_PROJECT_DIR)/src
 LIB_DIR            = $(WM_PROJECT_DIR)/lib
-LIB_WM_OPTIONS_DIR = $(LIB_DIR)/$(WM_OPTIONS)
+LIB_WM_OPTIONS_DIR = $(LIB_DIR)
 OBJECTS_DIR        = $(MAKE_DIR)/$(WM_OPTIONS)
 CLASSES_DIR        = $(MAKE_DIR)/classes
 
--- openfoam-1.6-ext-1.6.0.orig/src/fvAgglomerationMethods/MGridGenGamgAgglomeration/Make/options
+++ openfoam-1.6-ext-1.6.0/src/fvAgglomerationMethods/MGridGenGamgAgglomeration/Make/options
@@ -7,9 +7,9 @@ TYPE_REAL=-DTYPE_REAL
 
 EXE_INC = \
     -I$(LIB_SRC)/finiteVolume/lnInclude \
-    -I$(PARMGRIDGEN_DIR)/include/Lib    \
-    -I$(PARMGRIDGEN_DIR)/include/IMlib  \
+    -I/usr/include/mgridgen \
     $(TYPE_REAL)
 
 LIB_LIBS = \
-    -L$(PARMGRIDGEN_DIR)/lib -lMGridGen
+    -lMGridGen \
+    -lIMlib
--- openfoam-1.6-ext-1.6.0.orig/src/dynamicMesh/meshMotion/mesquiteMotionSolver/Make/options
+++ openfoam-1.6-ext-1.6.0/src/dynamicMesh/meshMotion/mesquiteMotionSolver/Make/options
@@ -1,9 +1,8 @@
 EXE_INC = \
-    -I$(MESQUITE_DIR)/include \
     -I$(LIB_SRC)/meshTools/lnInclude \
     -I$(LIB_SRC)/dynamicMesh/dynamicMesh/lnInclude \
 
 LIB_LIBS = \
     -lmeshTools \
     -ldynamicMesh \
-    -L$(MESQUITE_DIR)/lib -lmesquite
+    -lmesquite
--- openfoam-1.6-ext-1.6.0.orig/src/OpenFOAM/meshes/polyMesh/mapPolyMesh/objectMap/objectMapI.H
+++ openfoam-1.6-ext-1.6.0/src/OpenFOAM/meshes/polyMesh/mapPolyMesh/objectMap/objectMapI.H
@@ -123,18 +123,6 @@ inline Ostream& operator<<(Ostream& os, 
 }
 
 
-inline Istream& operator>>(Istream& is, objectMap& a)
-{
-    is.readBegin("objectMap");
-    is  >> a.index_ >> a.masterObjects_;
-    is.readEnd("objectMap");
-
-    // Check state of Istream
-    is.check("Istream& operator>>(Istream&, objectMap&)");
-
-    return is;
-}
-
 // * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * //
 
 } // Master namespace Foam
--- openfoam-1.6-ext-1.6.0.orig/src/OpenFOAM/meshes/polyMesh/mapPolyMesh/objectMap/objectMap.H
+++ openfoam-1.6-ext-1.6.0/src/OpenFOAM/meshes/polyMesh/mapPolyMesh/objectMap/objectMap.H
@@ -50,7 +50,6 @@ class objectMap;
 inline bool operator==(const objectMap& a, const objectMap& b);
 inline bool operator!=(const objectMap& a, const objectMap& b);
 inline Ostream& operator<<(Ostream&, const objectMap&);
-inline Istream& operator>>(Istream&, objectMap&);
 
 
 /*---------------------------------------------------------------------------*\
@@ -102,8 +101,6 @@ public:
     // IOstream Operators
 
         friend Ostream& operator<<(Ostream&, const objectMap&);
-
-        friend Istream& operator>>(Istream&, objectMap&);
 };
 
 
--- openfoam-1.6-ext-1.6.0.orig/src/decompositionMethods/metisDecomp/Make/options
+++ openfoam-1.6-ext-1.6.0/src/decompositionMethods/metisDecomp/Make/options
@@ -1,10 +1,11 @@
+include $(RULES)/mplib$(WM_MPLIB)
+
 EXE_INC = \
-    -I$(METIS_DIR)/include \
+    $(PINC) \
+    -I/usr/include/metis \
     -I../decompositionMethods/lnInclude \
     -I../scotchDecomp/lnInclude
 
 LIB_LIBS = \
     -L$(FOAM_LIBBIN)/dummy \
-    -L$(METIS_DIR)/lib -lmetis \
-    -L$(METIS_DIR)/lib -lGKlib
-
+    -lmetis
--- openfoam-1.6-ext-1.6.0.orig/src/decompositionMethods/parMetisDecomp/Make/options
+++ openfoam-1.6-ext-1.6.0/src/decompositionMethods/parMetisDecomp/Make/options
@@ -2,13 +2,13 @@ include $(RULES)/mplib$(WM_MPLIB)
 
 EXE_INC = \
     $(PFLAGS) $(PINC) \
-    -I$(PARMETIS_DIR)/include/ParMETISLib \
-    -I$(PARMETIS_DIR)/include \
+    -I/usr/include/parmetis \
+    -I/usr/include/metis \
     -I../decompositionMethods/lnInclude \
     -I../metisDecomp/lnInclude \
     -I../scotchDecomp/lnInclude
 
 LIB_LIBS = \
     -L$(FOAM_MPI_LIBBIN) \
-    -L$(PARMETIS_DIR)/lib -lmetis-parmetis \
-    -L$(PARMETIS_DIR)/lib -lparmetis
+    -lmetis \
+    -lparmetis
--- openfoam-1.6-ext-1.6.0.orig/src/decompositionMethods/scotchDecomp/Make/options
+++ openfoam-1.6-ext-1.6.0/src/decompositionMethods/scotchDecomp/Make/options
@@ -1,7 +1,7 @@
 EXE_INC = \
-    -I$(SCOTCH_DIR)/include \
+    -I/usr/include/scotch \
     -I../decompositionMethods/lnInclude
 
 LIB_LIBS = \
-    -L$(SCOTCH_DIR)/lib -lscotch \
-    -L$(SCOTCH_DIR)/lib -lscotcherrexit
+    -lscotch \
+    -lscotcherrexit
--- openfoam-1.6-ext-1.6.0.orig/etc/bashrc
+++ openfoam-1.6-ext-1.6.0/etc/bashrc
@@ -46,10 +46,11 @@ export FOAM_DEV=1
 #
 # Location of FOAM installation
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-foamInstall=$HOME/$WM_PROJECT
+# foamInstall=$HOME/$WM_PROJECT
 # foamInstall=~$WM_PROJECT
 # foamInstall=/usr/local/$WM_PROJECT
 # foamInstall=/opt/$WM_PROJECT
+foamInstall=/usr/lib
 #
 # END OF (NORMAL) USER EDITABLE PART
 ################################################################################
@@ -77,7 +78,7 @@ export WM_PROJECT_DIR=$WM_PROJECT_INST_D
 # Location of third-party software
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 #: ${WM_THIRD_PARTY_DIR=$WM_PROJECT_INST_DIR/ThirdParty-$WM_PROJECT_VERSION}; export WM_THIRD_PARTY_DIR
-: ${WM_THIRD_PARTY_DIR=$WM_PROJECT_DIR/ThirdParty}; export WM_THIRD_PARTY_DIR
+# : ${WM_THIRD_PARTY_DIR=$WM_PROJECT_DIR/ThirdParty}; export WM_THIRD_PARTY_DIR
 
 # Source files, possibly with some verbosity
 _foamSource()
@@ -139,7 +140,7 @@ export WM_COMPILER_LIB_ARCH=
 : ${WM_COMPILE_OPTION:=Opt}; export WM_COMPILE_OPTION
 
 # WM_MPLIB = | OPENMPI | MPICH | MPICH-GM | HPMPI | GAMMA | MPI | QSMPI
-: ${WM_MPLIB:=OPENMPI}; export WM_MPLIB
+: ${WM_MPLIB:=SYSTEMOPENMPI}; export WM_MPLIB
 
 
 # Run options (floating-point signal handling and memory initialisation)
@@ -298,7 +299,7 @@ cleanEnv=`$cleanProg "$PATH" "$foamOldDi
 cleanEnv=`$cleanProg "$LD_LIBRARY_PATH" "$foamOldDirs"` && LD_LIBRARY_PATH="$cleanEnv"
 
 #- Clean DYLD_LIBRARY_PATH
-if [[ "$WM_ARCH_BASE" == "darwin" ]] 
+if [ "$WM_ARCH_BASE" = "darwin" ]
 then
     cleanEnv=`$cleanProg "$DYLD_LIBRARY_PATH" "$foamOldDirs"` && DYLD_LIBRARY_PATH="$cleanEnv"
 fi
@@ -312,14 +313,14 @@ export PATH LD_LIBRARY_PATH MANPATH
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 _foamSource $WM_PROJECT_DIR/etc/settings.sh
-_foamSource $WM_PROJECT_DIR/etc/aliases.sh
+# _foamSource $WM_PROJECT_DIR/etc/aliases.sh
 
 
 # Source user setup files for optional packages
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 # _foamSource $WM_PROJECT_DIR/etc/apps/paraview/bashrc
 # _foamSource $WM_PROJECT_DIR/etc/apps/paraview3/bashrc
-_foamSource $WM_PROJECT_DIR/etc/apps/ensight/bashrc
+# _foamSource $WM_PROJECT_DIR/etc/apps/ensight/bashrc
 
 
 # Clean environment paths again. Only remove duplicates
--- openfoam-1.6-ext-1.6.0.orig/etc/cshrc
+++ openfoam-1.6-ext-1.6.0/etc/cshrc
@@ -45,10 +45,11 @@ setenv FOAM_DEV 1
 #
 # Location of FOAM installation
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-set foamInstall = $HOME/$WM_PROJECT
+# set foamInstall = $HOME/$WM_PROJECT
 # set foamInstall = ~$WM_PROJECT
 # set foamInstall = /usr/local/$WM_PROJECT
 # set foamInstall = /opt/$WM_PROJECT
+set foamInstall = /usr/lib
 #
 # END OF (NORMAL) USER EDITABLE PART
 ###############################################################################
@@ -131,7 +132,7 @@ setenv WM_COMPILE_OPTION Opt
 #setenv WM_COMPILE_OPTION Debug
 
 # WM_MPLIB = | OPENMPI | MPICH | MPICH-GM | HPMPI | GAMMA | MPI | QSMPI
-if ( ! $?WM_MPLIB ) setenv WM_MPLIB OPENMPI
+if ( ! $?WM_MPLIB ) setenv WM_MPLIB SYSTEMOPENMPI
 
 
 # Run options (floating-point signal handling and memory initialisation)
@@ -304,7 +305,7 @@ setenv MANPATH `$cleanProg "$MANPATH" "$
 # Source project setup files
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~
 _foamSource $WM_PROJECT_DIR/etc/settings.csh
-_foamSource $WM_PROJECT_DIR/etc/aliases.csh
+# _foamSource $WM_PROJECT_DIR/etc/aliases.csh
 
 # Source user setup files for optional packages
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
--- openfoam-1.6-ext-1.6.0.orig/etc/settings.csh
+++ openfoam-1.6-ext-1.6.0/etc/settings.csh
@@ -37,7 +37,7 @@ alias _foamAddPath 'set path=(\!* $path)
 alias _foamAddLib 'setenv LD_LIBRARY_PATH \!*\:${LD_LIBRARY_PATH}'
 
 # location of the jobControl directory
-setenv FOAM_JOB_DIR $WM_PROJECT_INST_DIR/jobControl
+setenv FOAM_JOB_DIR $HOME/$WM_PROJECT/jobControl
 
 # wmake configuration
 setenv WM_DIR $WM_PROJECT_DIR/wmake
@@ -52,18 +52,18 @@ setenv WM_DECOMP_LIBS "-lfaceDecompFinit
 
 # base configuration
 setenv FOAM_APP $WM_PROJECT_DIR/applications
-setenv FOAM_APPBIN $WM_PROJECT_DIR/applications/bin/$WM_OPTIONS
+setenv FOAM_APPBIN $WM_PROJECT_DIR/applications/bin
 setenv FOAM_LIB $WM_PROJECT_DIR/lib
 setenv FOAM_LIBBIN $WM_PROJECT_DIR/lib/$WM_OPTIONS
 setenv FOAM_SRC $WM_PROJECT_DIR/src
 
 # shared site configuration - similar naming convention as ~OpenFOAM expansion
-setenv FOAM_SITE_APPBIN $WM_PROJECT_INST_DIR/site/$WM_PROJECT_VERSION/bin/$WM_OPTIONS
-setenv FOAM_SITE_LIBBIN $WM_PROJECT_INST_DIR/site/$WM_PROJECT_VERSION/lib/$WM_OPTIONS
+setenv FOAM_SITE_APPBIN $WM_PROJECT_INST_DIR/site/$WM_PROJECT_VERSION/bin
+setenv FOAM_SITE_LIBBIN $WM_PROJECT_INST_DIR/site/$WM_PROJECT_VERSION/lib
 
 # user configuration
-setenv FOAM_USER_APPBIN $WM_PROJECT_USER_DIR/applications/bin/$WM_OPTIONS
-setenv FOAM_USER_LIBBIN $WM_PROJECT_USER_DIR/lib/$WM_OPTIONS
+setenv FOAM_USER_APPBIN $WM_PROJECT_USER_DIR/applications/bin
+setenv FOAM_USER_LIBBIN $WM_PROJECT_USER_DIR/lib
 
 # convenience
 setenv FOAM_TUTORIALS $WM_PROJECT_DIR/tutorials
--- openfoam-1.6-ext-1.6.0.orig/etc/settings.sh
+++ openfoam-1.6-ext-1.6.0/etc/settings.sh
@@ -47,7 +47,7 @@ _foamAddLib()
     while [ $# -ge 1 ]
     do
         export LD_LIBRARY_PATH=$1:$LD_LIBRARY_PATH
-	if [ "$WM_ARCH_BASE" == "darwin" ]	    
+	if [ "$WM_ARCH_BASE" = "darwin" ]
 	then
 	    export DYLD_LIBRARY_PATH=$1:$DYLD_LIBRARY_PATH
 	fi
@@ -57,7 +57,7 @@ _foamAddLib()
 
 
 # location of the jobControl directory
-export FOAM_JOB_DIR=$WM_PROJECT_INST_DIR/jobControl
+export FOAM_JOB_DIR=$HOME/$WM_PROJECT/jobControl
 
 # wmake configuration
 export WM_DIR=$WM_PROJECT_DIR/wmake
@@ -72,18 +72,18 @@ export WM_DECOMP_LIBS="-lfaceDecompFinit
 
 # base configuration
 export FOAM_APP=$WM_PROJECT_DIR/applications
-export FOAM_APPBIN=$WM_PROJECT_DIR/applications/bin/$WM_OPTIONS
+export FOAM_APPBIN=$WM_PROJECT_DIR/applications/bin
 export FOAM_LIB=$WM_PROJECT_DIR/lib
-export FOAM_LIBBIN=$WM_PROJECT_DIR/lib/$WM_OPTIONS
+export FOAM_LIBBIN=$WM_PROJECT_DIR/lib
 export FOAM_SRC=$WM_PROJECT_DIR/src
 
 # shared site configuration - similar naming convention as ~OpenFOAM expansion
-export FOAM_SITE_APPBIN=$WM_PROJECT_INST_DIR/site/$WM_PROJECT_VERSION/bin/$WM_OPTIONS
-export FOAM_SITE_LIBBIN=$WM_PROJECT_INST_DIR/site/$WM_PROJECT_VERSION/lib/$WM_OPTIONS
+export FOAM_SITE_APPBIN=$WM_PROJECT_INST_DIR/site/$WM_PROJECT_VERSION/bin
+export FOAM_SITE_LIBBIN=$WM_PROJECT_INST_DIR/site/$WM_PROJECT_VERSION/lib
 
 # user configuration
-export FOAM_USER_APPBIN=$WM_PROJECT_USER_DIR/applications/bin/$WM_OPTIONS
-export FOAM_USER_LIBBIN=$WM_PROJECT_USER_DIR/lib/$WM_OPTIONS
+export FOAM_USER_APPBIN=$WM_PROJECT_USER_DIR/applications/bin
+export FOAM_USER_LIBBIN=$WM_PROJECT_USER_DIR/lib
 
 # convenience
 export FOAM_TUTORIALS=$WM_PROJECT_DIR/tutorials
@@ -213,7 +213,7 @@ SYSTEMOPENMPI)
     export OPAL_PREFIX=
 
     # Set compilation flags here instead of in wmake/rules/../mplibSYSTEMOPENMPI
-    export PINC=`mpicc --showme:compile` 
+    export PINC="`mpicc --showme:compile`"
     export PLIBS="`mpicc --showme:link`"
     libDir=`echo "$PLIBS" | sed -e 's/.*-L\([^ ]*\).*/\1/'`
 
